enumerables.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def return_array(elements)
+    if self.class == Range
+      self
+    else
+      elements
+    end
+  end
+
+  def my_each
+    return to_enum unless block_given?
+
+    elements = to_a
+    elements.length.times do |i|
+      yield(elements[i])
+    end
+    return_array(elements)
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = to_a
+    elements.length.times do |i|
+      yield(elements[i], i)
+    end
+    return_array(elements)
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    elements = to_a
+    array = []
+    elements.my_each do |i|
+      array.push(i) if yield(i)
+    end
+    array
+  end
+
+  def my_all?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        return false unless i.class == Integer || i.class == Float || i.Complex
+      end
+      true
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false unless i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false unless i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false unless yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false unless i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false unless i == default
+      end
+      true
+    end
+  end
```

Comment:
### There are still some changes to be made in this method.

- [ ] When you pass a class as an argument for example let's say a class of numeric(numbers), the method should return true if all the members in the collection are of such a class.
### YOURS
![numeric yours](https://user-images.githubusercontent.com/32361963/96236420-b2938e80-0fa4-11eb-998d-27e3f8365171.png)
### ORIGINAL
![numeric  original](https://user-images.githubusercontent.com/32361963/96236454-bc1cf680-0fa4-11eb-84ef-7c57d3b5e237.png)